src/OrbitBase/include/OrbitBase/File.h

```diff
@@ -13,11 +13,52 @@
 
 #include "OrbitBase/Result.h"
 #include "OrbitBase/SafeStrerror.h"
-#include "OrbitBase/UniqueResource.h"
+
+#if defined(__linux)
+#include <unistd.h>
+#elif defined(_WIN32)
+#include <io.h>
+#endif
 
 namespace orbit_base {
 
-using unique_fd = unique_resource<int, void (*)(int)>;
+constexpr int kInvalidFd = -1;
+
+class unique_fd {
+ public:
+  constexpr unique_fd() noexcept = default;
+  constexpr explicit unique_fd(int fd) noexcept : fd_{fd} {}
+  ~unique_fd() noexcept { release(); }
+
+  unique_fd(const unique_fd&) = delete;
+  unique_fd& operator=(const unique_fd&) = delete;
+
+  constexpr unique_fd(unique_fd&& other) noexcept : fd_{other.fd_} { other.fd_ = kInvalidFd; }
+
+  unique_fd& operator=(unique_fd&& other) noexcept {
+    reset(other.fd_);
+    other.fd_ = kInvalidFd;
```

Comment:
I think you need to guard againat self-move!

Reply:
right, thanks!

Reply:
done, and added a test.